pkg/scheduler/framework/parallelize/parallelism.go

```diff
@@ -51,15 +51,29 @@ func chunkSizeFor(n, parallelism int) int {
 	return s
 }
 
+// workersForChunkSize returns number of workers (goroutines)
+// that will be created in workqueue.ParallelizeUntil
+// for given parallelism, pieces and chunkSize values.
+func workersForChunkSize(parallelism, pieces, chunkSize int) int {
+	chunks := (pieces + chunkSize - 1) / chunkSize
+	if chunks < parallelism {
+		return chunks
+	}
+	return parallelism
+}
+
 // Until is a wrapper around workqueue.ParallelizeUntil to use in scheduling algorithms.
 // A given operation will be a label that is recorded in the goroutine metric.
 func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc, operation string) {
 	goroutinesMetric := metrics.Goroutines.WithLabelValues(operation)
 	withMetrics := func(piece int) {
-		goroutinesMetric.Inc()
 		doWorkPiece(piece)
 		goroutinesMetric.Dec()
```

Comment:
According to the implementation of `ParallelizeUntil`, won't `Dec()` called more than the worker num calculated from `workersForChunkSize()`?
- `workersForChunkSize()` equals to the number of goroutines. 
  - IIUC, it returns the same number as `workers` [here](https://github.com/kubernetes/kubernetes/blob/95d71c464a6a5f81ae36bddc86972b3b2a1c40a5/staging/src/k8s.io/client-go/util/workqueue/parallelizer.go#L75).
- `goroutinesMetric.Dec()` could be called multiple times within goroutines.
  - https://github.com/kubernetes/kubernetes/blob/95d71c464a6a5f81ae36bddc86972b3b2a1c40a5/staging/src/k8s.io/client-go/util/workqueue/parallelizer.go#L90

Reply:
It seems it is equal to chunks instead of workers.

Reply:
Ouch, that's right. So now, I'm not sure if my change is possible easily. Before, `Inc()` and `Dec()` were called `pieces` number of times and were capped by number of `workers`. We can't simply do `Add(pieces)`, because we would register much more goroutines than we have. 

I'm wondering if we couldn't do batch `Add(-workers)` (similar to introduced `Add(workers)`) instead of decrementing per each work piece. It won't be fully accurate, but still will give us the most important semantic, which is finding leaked goroutines.

Reply:
Please check the actual implementation